src/lib/components/composites/settings/project-settings/general/ProjectNameSettings.svelte

```diff
@@ -0,0 +1,124 @@
+<script lang="ts">
+    import SettingsSection from "$lib/components/composites/settings/SettingsSection.svelte";
+    import Input from "$lib/components/composites/input/Input.svelte";
+    import { Button } from "$lib/components/primitives/button";
+    import { backendService } from "$lib/grpc-api";
+    import { onMount } from "svelte";
+    import { Project } from "$lib/model/api/project";
+    import { Schema } from "$lib/schemas";
+    import { generateFieldMask } from "protobuf-fieldmask";
+    import type { ApiError } from "$lib/model/general";
+    import { StatusCodes } from "$lib/model/error-codes";
+    import ErrorAlert from "$lib/components/composites/utils/ErrorAlert.svelte";
+
+    interface Props {
+        projectId: string,
+        loadingProject: Promise<Project>;
+    }
+
+    const { projectId, loadingProject }: Props = $props();
+
+    let projectName: string = $state("");
+
+    let projectNameInput: Input;
+
+
+    let updateProjectError: ApiError | undefined = $state(undefined);
+
+    /**
+     * Loading the project name on mount.
+     */
+    onMount(async () => {
+        loadingProject
+            .then((project) => {
+                projectName = project.name;
+            })
+            .catch((error) => {
+                console.error(error);
+            })
+    });
+
+    /**
+     * Changes the name of the project, if a non-blank project name is provided.
+     *
+     * Therefore, send a request to the backend to update the project with the
+     * provided name.
+     */
+    async function handleSubmit(event: Event) {
+        event.preventDefault();
+
+        const projectNameValid = projectNameInput.validate();
+        if (!(projectNameValid)) return;
+
+        const projectData: Partial<Project> = {
+            id: projectId,
+            name: projectNameInput.getValue(),
+        };
+
+        const fieldMask = generateFieldMask(projectData);
+        console.log("fieldMask", fieldMask);
+
+        backendService
+            .updateProject({
+                project: Project.create(projectData),
+                mask: {
+                    paths: fieldMask,
+                },
+            })
+            .then(() => {return})
+            .catch((error) => {
+                console.log(error.code)
+                if (error.code === StatusCodes.ALREADY_EXISTS) {
+                    updateProjectError = {
+                        errorTitle: "TODO",
+                        errorDetails: "TODO",
+                    }
+                } else {
+                    updateProjectError = { errorTitle: "Somethin went wrong while updating project name."};
+                }
+                console.error(error);
+            });
+    }
+
+</script>
+
+<!--
+@component
+Component for changing the project name settings.
+
+Usage:
+```svelte
+    <SettingsSection />
```

Comment:
Provide a correct example how this component can be used.
```suggestion
    <ProjectNameSettings {loadingProject} {projectId}/>
```